### DISCUSSION TITLE
Read existing zonebudget6 output with flopy.mf6.utils

### DISCUSSION THREAD

**Discussion by acmoody** [modflowpy/flopy#2328] [Q&A]
I was hoping I could use flopy zonebudget6 methods to read an existing zone budget output csv without accessing the output through `gwf.output.zonebudget`. I haven't seen any examples showing how to do this (maybe because it is discouraged) and my approach does not work, which is 

```python
>>> zb = fp.mf6.utils.output_util.ZoneBudget6('BigRch', model_ws=path/to/zboutput, extension='zbnam')
>>> zb.get_dataframes()
AttributeError: 'NoneType' object has no attribute '_zonenamedict'
```
My path has the izone array and zbnam file. I could use pandas and deal with the time myself, but if this can be accomplished with flopy, I'd be interested to know how.

**Comment by jlarsen-usgs** [comment]
@acmoody 

The `mf6.utils.output_util` module is used for generating dynamic `.output` attributes on the model and package object. The `ZoneBudget6` class doesn't actually live here and is imported for use by the `MF6Output` class. 

Try using this instead:

```python
zb = fp.utils.zonbud.ZoneBudget6.load('BigRch', model_ws=path/to/zboutput)
recarray = zb.get_budget(f="optional output file name")
df = zb.get_dataframes()
```

**Reply by acmoody** [reply]
That seems more straightforward. I'm now running into this error in flopy 3.8.1

```python
Cell In[102], line 1
    fp.utils.zonbud.ZoneBudget6.load('BigRch.zbnam',model_ws=temp_wd)

  File ~\Anaconda3\envs\theis\lib\site-packages\flopy\utils\zonbud.py:2120 in load
    zb6.add_package(t[0], t[1])

  File ~\Anaconda3\envs\theis\lib\site-packages\flopy\utils\zonbud.py:1885 in add_package
    pkg = func.load(pkg, self)

  File ~\Anaconda3\envs\theis\lib\site-packages\flopy\utils\zonbud.py:2298 in load
    izone[i] = zn

ValueError: invalid literal for int() with base 10: ' '
```

What appears to be happening is `multilinestrip(fobj)` is returning a string `t = '0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0'`  in line 2292 in `zonbud.py` and then that string is iterated over, white space included, and izone[i] can't be set to `' '`

**Reply by jlarsen-usgs** [reply]
@acmoody 

Can you zip up and attach your zonebudget input and output files. I'll take a look and see what's going on and if there is a bug somewhere that needs fixing

**Reply by acmoody** [reply]
[zb.zip](https://github.com/user-attachments/files/17285050/zb.zip)
GitHub didn't like the size of the zip with the budget file included. Can you  make do without?
